src/components/modal/ModalHeader.tsx

```diff
@@ -21,15 +21,19 @@ export interface IModalHeaderDispatchProps {
 export interface IModalHeaderProps extends IModalHeaderOwnProps, IModalHeaderStateProps, IModalHeaderDispatchProps {}
 
 export class ModalHeader extends React.Component<IModalHeaderProps, {}> {
-    private canClose: boolean = true;
+    static defaultProps: Partial<IModalHeaderProps> = {
+        lastOpened: true,
+    };
+
+    private canClose: boolean;
 
     componentDidMount() {
-        this.canClose = _.result(this.props, 'lastOpened', true);
+        this.canClose = this.props.lastOpened;
     }
 
     componentDidUpdate() {
         this.canClose = false;
-        setTimeout(() => this.canClose = _.result(this.props, 'lastOpened', true), 1);
+        _.defer(() => this.canClose = this.props.lastOpened);

```

Comment:
`setTimeout(() => this.canClose = this.props.lastOpened);` would act the same here unless i'm wrong, _.defer uses setTimeout without a time specified under the hood, and if not specified, setTimeout uses `4` by default in most browsers https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout see `Timeouts throttled to >=4ms` that being said both are fine